packages/connect/src/protocol-connect/end-stream.ts

```diff
@@ -21,7 +21,7 @@ import { errorFromJson, errorToJson } from "./error-json.js";
 import { appendHeaders } from "../http-headers.js";
 import { ConnectError } from "../connect-error.js";
 import { Code } from "../code.js";
-import type { Serialization } from "../protocol";
+import type { Serialization } from "../protocol/serialization.js";
```

Comment:
This fixes an [Internal Resolution Error](https://github.com/arethetypeswrong/arethetypeswrong.github.io/blob/main/docs/problems/InternalResolutionError.md) with Node 16.

Reply:
Reported by `attw` I assume?

I can reproduce with:

<details open>
<summary>tsconfig.json</summary>

```json
{
  "files": ["test.ts"],
  "compilerOptions": {
    "module": "node16",
    "skipLibCheck": false
  }
}
```
</details>


<details open>
<summary>test.ts</summary>

```ts
import {endStreamFlag} from "@connectrpc/connect/protocol-connect";

console.log(endStreamFlag)
```
</details>

```bash
$ npx tsc                                                                                    
../connect/dist/esm/protocol-connect/end-stream.d.ts:3:36 - error TS2834: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Consider adding an extension to the import path.

3 import type { Serialization } from "../protocol";
                                     ~~~~~~~~~~~~~


Found 1 error in ../connect/dist/esm/protocol-connect/end-stream.d.ts:3
```

Nice, this is a genuine issue found by `attw`. 

The doc you linked provides further guidance:

> Library authors should always compile/check with --module node16 --moduleResolution node16 or nodenext (note that module implies the corresponding moduleResolution value, but not vice versa), because this is the only TypeScript mode that recognizes Node’s strict module resolution algorithm for ESM.

The `Node` module resolution algorithm we're using is effectively Node v10. I think it's a good idea to switch to `node16` across the board, but I believe this PR is much easier to reason about if we do it separately (separate PR).

Reply:
It was uncovered by `attw`, yeah. And yeah, sounds good re: `node16`. We can def do that in a follow-up PR.